darwin/importer/importer.py

```diff
@@ -952,36 +973,55 @@ def import_annotations(  # noqa: C901
         ]
         if files_to_track:
             _warn_unsupported_annotations(files_to_track)
-            for parsed_file in track(files_to_track):
-                image_id, default_slot_name = remote_files[parsed_file.full_path]
-                # We need to check if name is not-None as Darwin JSON 1.0
-                # defaults to name=None
-                if parsed_file.slots and parsed_file.slots[0].name:
-                    default_slot_name = parsed_file.slots[0].name
-
-                metadata_path = is_properties_enabled(parsed_file.path)
-
-                errors, _ = _import_annotations(
-                    dataset.client,
-                    image_id,
-                    remote_classes,
-                    attributes,
-                    parsed_file.annotations,  # type: ignore
-                    default_slot_name,
-                    dataset,
-                    append,
-                    delete_for_empty,
-                    import_annotators,
-                    import_reviewers,
-                    metadata_path,
-                )
 
-                if errors:
-                    console.print(
-                        f"Errors importing {parsed_file.filename}", style="error"
-                    )
-                    for error in errors:
-                        console.print(f"\t{error}", style="error")
+            if use_multi_cpu:
+                with concurrent.futures.ThreadPoolExecutor(
+                    max_workers=cpu_limit
+                ) as executor:
+                    futures = [
+                        executor.submit(import_annotation, file)
+                        for file in files_to_track
+                    ]
```

Comment:
There are 2 multi-threading loops (1 inside the other). Both are necessary. This is the inner loop, and it parallelises the import of each `import_annotation` object

This is useful for annotation formats where 1 JSON file can represent multiple dataset items, such as COCO, where 1 file contains the entire dataset. In these cases, you could have a large number of `import_annotation` objects in `files_to_track`